core/tests/test_views.py

```diff
@@ -117,6 +164,61 @@ def test_add_comment_unauthorized(client: Client, post: Post) -> None:
     assert reverse("login") in response.url
 
 
+def test_reported_list_post_by_admin(client: Client, admin: User, post: Post, report_data: dict) -> None:
+    client.force_login(admin)
+    response = client.post(reverse("post-report", kwargs={"pk": post.pk}), data=report_data)
```

Comment:
This submits post report right?
So we are testing here several things
- creating post report (because you sent HTTP request)
- redirections
- and finally post list.

This is not only huge test, it is also slow because it does too many things. It is not Unit test, this is integration test. 

It would be much better to create the report on DB level using objects create. Or easier: run the site, create post and report manually, use "dumpdata" command to dump fixture into file and load that fixture into DB duiring tests. Details below. 
Keep it simple!

---  

To load JSON fixtures into the database for testing with pytest, you can use the following approach:

1. Create a JSON fixture file with your test data.

2. Use a pytest fixture to load the data into the database before running tests.

Here's how you can do it:

1. Create a JSON fixture file (e.g. `test_data.json`) with your test data:

```json
[
  {
    "model": "myapp.MyModel",
    "pk": 1,
    "fields": {
      "field1": "value1",
      "field2": "value2"
    }
  },
  {
    "model": "myapp.MyModel",
    "pk": 2,
    "fields": {
      "field1": "value3",
      "field2": "value4"
    }
  }
]
```

2. Create a pytest fixture in your `conftest.py` file to load the fixture data:

```python
import pytest
from django.core.management import call_command

@pytest.fixture(scope='session')
def django_db_setup(django_db_setup, django_db_blocker):
    with django_db_blocker.unblock():
        call_command('loaddata', 'test_data.json')
```

This fixture does the following:
- It has session scope, so it runs once for the entire test session.
- It depends on `django_db_setup` to ensure the test database is set up.
- It uses `django_db_blocker.unblock()` to allow database access.
- It calls Django's `loaddata` management command to load the JSON fixture.

3. Use the fixture in your tests:

```python
@pytest.mark.django_db
def test_my_model(django_db_setup):
    from myapp.models import MyModel
    
    # The fixture data is now loaded in the test database
    assert MyModel.objects.count() == 2
    assert MyModel.objects.filter(field1="value1").exists()
```

By using the `django_db_setup` fixture in your test, you ensure that the JSON data is loaded into the test database before your test runs.

Remember to place your JSON fixture file in a location where Django can find it, typically in a `fixtures` directory within your app.

This approach allows you to load your JSON test data into the database for your pytest tests, providing a consistent starting point for your database-dependent tests.

Citations:
[1] https://pytest-with-eric.com/pytest-best-practices/pytest-read-json/
[2] https://flowfx.de/blog/populate-your-django-test-database-with-pytest-fixtures/
[3] https://pytest-django.readthedocs.io/en/latest/database.html
[4] https://stackoverflow.com/questions/50329629/how-to-access-a-json-filetest-data-like-config-json-in-conftest-py
[5] https://stackoverflow.com/questions/57702637/how-to-parametrize-tests-with-json-array-test-data-using-pytest-in-python
[6] https://automationpanda.com/2023/01/12/passing-test-inputs-into-pytest/
[7] https://www.testcult.com/handle-test-data-the-right-way-in-pytest/
[8] https://www.geeksforgeeks.org/read-json-file-using-python/
[9] https://github.com/pytest-dev/pytest-django/issues/13
[10] https://realpython.com/django-pytest-fixtures/
[11] https://www.freecodecamp.org/news/python-read-json-file-how-to-load-json-from-a-file-and-parse-dumps/
[12] https://pypi.org/project/pytest-json-fixtures/
[13] https://opensource.com/article/19/7/save-and-load-data-python-json
[14] https://code-maven.com/slides/python/pytest-create-fixture-with-files-fixture
[15] https://pytest-django.readthedocs.io/en/stable/database.html
[16] https://docs.pytest.org/en/7.1.x/builtin.html

Reply:
I don't know how to do it because I'm trying with chat but it doesn't work